### PR TITLE
Support randomized passive listening ports

### PR DESCRIPTION
**Pull request by robklg**

### CODE REVIEW

File: src/server/mod.rs [hunk 1]
```diff
@@ -419,7 +423,24 @@ where
...
                             ensure_authenticated!();
 
-                            let listener = std::net::TcpListener::bind(&passive_addrs.as_slice())?;
+                            let mut rng = rand::thread_rng();
+
+                            let mut listener: Option<std::net::TcpListener> = None;
+                            for _ in 1..BIND_RETRIES {
+                                let i = rng.gen_range(0, passive_addrs.len()-1);
+                                match std::net::TcpListener::bind(passive_addrs[i]) {
+                                    Ok(x) => {
```
Comment (mdirkse) [inline src/server/mod.rs]: Don't need the parentheses as it's a one-liner (like the other branch)

File: src/server/mod.rs [hunk 2]
```diff
@@ -419,7 +423,24 @@ where
...
+                                    }
+                                    Err(_) => continue,
+                                };
+                            };
+
+                            if listener.is_none() {
+                                return Ok(Reply::new(ReplyCode::CantOpenDataConnection, "No data connection established"));
+                            }
+
+                            let listener = listener.unwrap();
```
Comment (mdirkse) [inline src/server/mod.rs]: can combine this with the previous if into a match statement that returns on None and sets listener on Some

Something like:
`let listener = match listener {
    None => return Ok(Reply::new(ReplyCode::CantOpenDataConnection, "No data connection established")),
    Some(l) => l.unwrap(),
};`

File: src/server/mod.rs [hunk 3]
```diff
@@ -419,7 +423,24 @@ where
                         Command::Pasv => {
                             ensure_authenticated!();
 
-                            let listener = std::net::TcpListener::bind(&passive_addrs.as_slice())?;
+                            let mut rng = rand::thread_rng();
+
+                            let mut listener: Option<std::net::TcpListener> = None;
+                            for _ in 1..BIND_RETRIES {
```
Comment (mdirkse) [inline src/server/mod.rs]: Does the loop end when an open port is found?

Comment (robklg) [reply src/server/mod.rs]: right... ofcourse :-S

File: src/server/mod.rs (hunk 2 above)
Comment (robklg) [reply src/server/mod.rs]: good one

File: src/server/mod.rs (hunk 3 above)
Comment (robklg) [reply src/server/mod.rs]: added break statement

**Comment by mdirkse** [comment]
LGTM

**Pull request merged**